File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        incNum = 1
        decNum = 1
        maxLen = 1

        for i in range(len(nums) - 1):
            if nums[i] > nums[i + 1]:
                decNum += 1
                incNum = 1
            elif nums[i] < nums[i + 1]:
                incNum += 1
                decNum = 1
            else:
                 incNum = 1
                 decNum = 1
            maxLen = max(maxLen, incNum, decNum)
        return maxLen

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestMonotonicSubarray = function(nums) {
    let maxLen = 1
    let decNum = 1
    let incNum = 1

    for(let i = 0; i < nums.length; i++){
        if(nums[i] > nums[i + 1]){
            decNum += 1
            incNum = 1
        }
        else if(nums[i] < nums[i + 1]){
            incNum += 1
            decNum = 1
        }
        else{
            incNum = 1
            decNum = 1
        }
        maxLen = Math.max(maxLen,incNum,decNum)
    }
    return maxLen
};